src/pyflame.cc

```diff
@@ -139,24 +194,16 @@ int main(int argc, char **argv) {
         std::this_thread::sleep_for(interval);
         PtraceAttach(pid);
       }
-      if (idle) {
-        std::cout << "(idle) " << idle << "\n";
-      }
-      // process the frames
-      for (const auto &kv : buckets) {
-        if (kv.first.empty()) {
-          std::cerr << "uh oh\n";
+      if (!include_ts) {
+        if (0 != print_frames(idle, buckets)) {
           return 1;
         }
-        auto last = kv.first.rend();
-        last--;
-        for (auto it = kv.first.rbegin(); it != last; ++it) {
-          std::cout << *it << ";";
-        }
-        std::cout << *last << " " << kv.second << "\n";
       }
     } else {
       const unsigned long frame_addr = FirstFrameAddr(pid, tstate_addr);
+      if (include_ts) {
```

Comment:
I think it would be better to just accumulate the results in a vector (regardless of which mode we're in) and then just have a loop that either turns the vector into a map for the pyflame case, or prints out everything in the vector in the timestamp case.